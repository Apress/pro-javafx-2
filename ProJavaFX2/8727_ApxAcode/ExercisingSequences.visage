/*
 * @test
 * @run
 */

var booleans = [true, false, true, true, false];
var integers = [1, 3, 5, 7, 9];
var numbers = [3.14159, 2.71828];
var strings = ["hello", "hello again", "goodbye"];
var durations = [0.5m, 1.0m, 1.5m, 2.0m];

print("booleans = "); println(booleans);
print("integers = "); println(integers);
print("numbers = "); println(numbers);
print("strings = "); println(strings);
print("durations = "); println(durations);

class Point {
  var x: Number;
  var y: Number;
  override function toString() {
    "Point \{ x: {x}, y: {y} \}"
  }
}

var points = [Point {x: 3.0, y: 4.0}, Point {x: 5.0, y: 12.0}];

print("points = "); println(points);

integers = [1, 3, 5, 7, 9];
print("integers = "); println(integers);

var oneToTen = [1..10];
var oneToTenOdd = [1..10 step 2];
var ticks = [3.0 .. 5.0 step 0.5];
var decreasing = [10..1 step -1];
var oneToNine = [1 ..< 10];

print("oneToTen = "); println(oneToTen);
print("oneToTenOdd = "); println(oneToTenOdd);
print("ticks = "); println(ticks);
print("decreasing = "); println(decreasing);
print("oneToNine = "); println(oneToNine);

print("[10..1] = "); println([10..1]);
