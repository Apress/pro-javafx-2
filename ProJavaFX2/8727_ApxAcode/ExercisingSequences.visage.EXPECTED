booleans = [ true, false, true, true, false ]
integers = [ 1, 3, 5, 7, 9 ]
numbers = [ 3.14159, 2.71828 ]
strings = [ hello, hello again, goodbye ]
durations = [ 30000.0ms, 60000.0ms, 90000.0ms, 120000.0ms ]
points = [ Point { x: 3.0, y: 4.0 }, Point { x: 5.0, y: 12.0 } ]
integers = [ 1, 3, 5, 7, 9 ]
oneToTen = [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
oneToTenOdd = [ 1, 3, 5, 7, 9 ]
ticks = [ 3.0, 3.5, 4.0, 4.5, 5.0 ]
decreasing = [ 10, 9, 8, 7, 6, 5, 4, 3, 2, 1 ]
oneToNine = [ 1, 2, 3, 4, 5, 6, 7, 8, 9 ]
[10..1] = [ ]
